clmm/utils/validation.py

```diff
@@ -207,3 +207,20 @@ def _validate_dec(loc, dec_name, is_array):
     """
     v_type = "float_array" if is_array else (float, str)
     validate_argument(loc, dec_name, v_type, argmin=-90, eqmin=True, argmax=90, eqmax=True)
+
+
+def _validate_is_deltasigma_sigma_c(is_deltasigma, sigma_c):
+    r""" "Validate the compatibility between is_deltasigma and sigma_c arguments.
+
+
+    Parameters
+    ----------
+    is_deltasigma: bool
+        If `False`, values are computed for shear, else they are computed for :math:`\Delta \Sigma`.
+    sigma_c : None, array_like
+        Critical (effective) surface density in units of :math:`M_\odot\ Mpc^{-2}`.
+    """
+    if is_deltasigma and sigma_c is None:
+        raise TypeError("sigma_c (=None) must be provided when is_deltasigma=True")
+    if not is_deltasigma and sigma_c is not None:
+        raise TypeError(f"sigma_c (={sigma_c}) be provided when is_deltasigma=False")
```

Comment:
I'd suggest only warning the user that sigma_c is ignored if is_deltasigma=False

```
    if not is_deltasigma and sigma_c is not None:
        warnings.warn(f"sigma_c (={sigma_c}) is ignored when is_deltasigma=False")

```